src/api/java/com/minecolonies/api/crafting/IItemStorageFactory.java

```diff
@@ -44,8 +44,8 @@ default ItemStorage getNewInstance(@NotNull final IFactoryController factoryCont
 
         final ItemStack stack = (ItemStack) context[0];
         final int size = (int) context[1];
-        final boolean ignoreDamage = context[2] != null ? (Boolean) context[2] : false;
-        final boolean ignoreNBT = context[3] != null ? (Boolean) context[3] : false;
+        final boolean ignoreDamage = context.length >= 3 ? (Boolean) context[2] : false;
+        final boolean ignoreNBT = context.length >= 4 ? (Boolean) context[3] : false;
```

Comment:
This is better, but it's inconsistent with the comparisons above.  You should make both the same.  (Personally, I'd use the plain number, not the named constant, but that's up to you.)